libraries/AP_HAL/Scheduler.h

```diff
@@ -115,6 +115,12 @@ class AP_HAL::Scheduler {
         return false;
     }
 
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
```

Comment:
This is messy showing up here for all vehicle. It's a bit less type safe I admit, but I'd rather see this show up as a public member of the HAL_SITL scheduler, and see us cast down where we need it.

Reply:
It's all in SITL now.